rmk/src/config/keyboard_macros/keyboard_macro.rs

```diff
@@ -0,0 +1,412 @@
+use num_enum::FromPrimitive;
+
+use crate::{
+    keycode::KeyCode,
+    keymap::fill_vec,
+    via::keycode_convert::{from_ascii, to_ascii},
+};
+
+use crate::config::keyboard_macros::macro_config::MACRO_SPACE_SIZE;
+
+/// encoded with the two bytes, content at the third byte
+/// 0b 0000 0001 1000-1010 (VIAL_MACRO_EXT) are not supported
+///
+/// TODO save space: refacter to use 1 byte for encoding and convert to/from vial 2 byte encoding
+#[derive(Debug, Clone)]
+pub enum MacroOperation {
+    /// 0x00, 1 byte
+    /// Marks the end of a macro sequence
+    /// Don't use it on your own,
+    /// will be automatically removed and added
+    /// by MacroOperations::define_macro_sequences()
+    End,
+    /// 0x01 01 + 1 byte keycode
+    Tap(KeyCode),
```

Comment:
`Tap(KeyCode::A)` is converted to `[0x01, 0x01, 0x04]`, am I right?

Why don't use 0x01 for `Tap`, 0x02 for `Press` and so on?